pilot/cmd/pilot-discovery/debug.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/pkg/log"
+)
+
+type debug struct {
+	pilotAddress string
+}
+
+var (
+	configTypes = map[string]string{
+		"all":       "",
+		"clusters":  "cdsz",
+		"listeners": "ldsz",
+		"routes":    "rdsz",
+		"endpoints": "edsz",
+	}
+
+	debugCmd = &cobra.Command{
+		Use:   "debug <proxyID> <configuration-type>",
+		Short: "Retrieve the configuration for the specified proxy",
+		Args:  cobra.MinimumNArgs(2),
+		RunE: func(c *cobra.Command, args []string) error {
+			d := &debug{
+				pilotAddress: "127.0.0.1:15007",
+			}
+			return d.run(args)
+		},
+	}
+)
+
+func (d *debug) run(args []string) error {
```

Comment:
General question: why an executable rather than just a utility method?  Regardless, it would seem that we could put a lot of the code from this PR into a `pilot_util.go` (name?) .. if we really want an executable, it could just be a thin wrapper around the utility methods. WDYT?

Reply:
Could you give an example of what you mean by a utility method? The reason I went with putting this in the binary is to ensure it works in FROM scratch Pilots. If there is a work around I'm missing for this issue let me know!

Reply:
I guess what I was thinking was that all of this boils down to the `printConfig` method, which is really just a client-side utility method.  Couldn't we just make that a public method that could be called from anywhere (all you need is the pilot address)? What's the advantage to having this code on the pilot pod?

Reply:
Correct, my point is that it seems to be a general utility method.  I'm fine leaving it as-is for now.